DM-49646: Update FBS scheduler configuration for small-field science programs for LSSTCam

bechtol opened the pull request:
This PR updates FBS scheduler configuration for small-field science programs for LSSTCam.

Example [notebook](https://github.com/lsst-sitcom/notebooks_svv_survey_planning/blob/main/notebooks/lsstcam/DM-49646.ipynb) used for testing:

rhiannonlynne commented on Scheduler/feature_scheduler/maintel/fbs_config_minimal_fieldsurvey.py:
```diff
@@ -67,60 +88,31 @@ def get_scheduler():
-        basis_functions.SlewtimeBasisFunction(filtername="z_03", nside=nside),
-        basis_functions.SlewtimeBasisFunction(filtername="y_04", nside=nside),
-        basis_functions.FilterLoadedBasisFunction(filternames=sequence),
+        basis_functions.SlewtimeBasisFunction(bandname="u", nside=nside),
```
We did add the option to have the SlewtimeBasisFunction be bandname = None, then it just returns the slewtime without considering a filter change (which is what you want I think). 
See https://rubin-scheduler.lsst.io/fbs-api-basis-functions.html#rubin_scheduler.scheduler.basis_functions.SlewtimeBasisFunction

I would suggest consolidating all of these SlewtimeBasisFunctions to a single one with "None" as the bandname.

rhiannonlynne commented on Scheduler/feature_scheduler/maintel/fbs_config_minimal_fieldsurvey.py:
```diff
@@ -67,60 +88,31 @@ def get_scheduler():
+        basis_functions.SlewtimeBasisFunction(bandname="z", nside=nside),
+        basis_functions.SlewtimeBasisFunction(bandname="y", nside=nside),
+        # TODO: check if BandLoadedBasisFunction is needed
+        basis_functions.BandLoadedBasisFunction(bandnames=sequence),
```
You should NOT use this basis function if you want the sequence to execute even if one of the bands is unavailable. If you want the sequence to fail is one of the bands is unavailable, leave it in.

rhiannonlynne commented:
I suspect you might want to extend the shadow minutes value for the AltAzShadowMaskBasisFunction, to be a bit closer to the length of the sequence you expect to acquire. 
Remember that the sequence will contain all of the bands in the requested sequence, minus whatever bands are not currently mounted.

bechtol commented:
> I suspect you might want to extend the shadow minutes value for the AltAzShadowMaskBasisFunction, to be a bit closer to the length of the sequence you expect to acquire. Remember that the sequence will contain all of the bands in the requested sequence, minus whatever bands are not currently mounted.

Extended the shadow minutes value. For the first tests, I think we will want to keep the sequences short to help debug any issues around the filter changes.

rhiannonlynne commented:
Hi Keith, I think you still have a survey configuration or two that is trying to import `get_comcam_sv_targets` but that this function got renamed to `get_sv_targets` ?  You will have to update the other configurations as well, or keep both functions available in ts_fbs_utils, or deprecate those survey configurations (not sure if there's a process around that).

bechtol commented:
> Hi Keith, I think you still have a survey configuration or two that is trying to import `get_comcam_sv_targets` but that this function got renamed to `get_sv_targets` ? You will have to update the other configurations as well, or keep both functions available in ts_fbs_utils, or deprecate those survey configurations (not sure if there's a process around that).

Removed the deprecated scheduler configuration. Updated the test cases in corresponding ts_fbs_utils PR.

edennihy reviewed (approved):
A few comments for your consideration, otherwise approved. However, I would like to test this on the BTS before merging, so please do not merge yet. 

We should merge the branch in ts_fbs_utils first, then test this on the BTS which is currently being upgraded with Cycle 41 and will have rubin_scheduler v3.6.0 (includes the band_to_filter translation).
